README.md

```diff
@@ -1,20 +1,236 @@
+
+## Preview and Debugging Rules
+
+Rule writing can sometimes take a few attempts to get it right.  Fortunately, the following two options can help.
+
+- When the configuration option `_dlo_previewMode` is set to `true`, output will be written to `console.log` rather than the `destination`.
+- When a particular rule's `debug` property is set to `true`, the incremental transformations performed by `operators` and additional logging will be written to `console.debug`.
+
+Viewing the JavaScript console with the above option set to `true` prints the following example.
+
+```javascript
+dataLayer.transaction[!(items)] handleData entry
+[{'transactionID':'2325112','total':{'grandTotal':'45.59','subTotal':'37.02','shippingTotal':'5.99','totalTax':'2.58','promotionCodes':[],'selectedPayment':'VISA'}}]
+dlo-debug.js:1136   [0] flatten output (numKeys=7 sizeOfValues=94 sizeOfPayload=342)
+  [{'transactionID':'2325112','grandTotal':'45.59','subTotal':'37.02','shippingTotal':'5.99','totalTax':'2.58','promotionCodes':[],'selectedPayment':'VISA'}]
+dlo-debug.js:1136   [1] convert output (numKeys=10 sizeOfValues=114 sizeOfPayload=434)
+  [{'transactionID':'2325112','grandTotal':45.59,'subTotal':37.02,'shippingTotal':5.99,'totalTax':2.58,'promotionCodes':[],'selectedPayment':'VISA','deliveryTotal':0,'shippingTax':0,'totalAdjustment':0}]
+dlo-debug.js:1136   [2] insert output (numKeys=0 sizeOfValues=0 sizeOfPayload=34)
+  ['Order Completed',{'transactionID':'2325112','grandTotal':45.59,'subTotal':37.02,'shippingTotal':5.99,'totalTax':2.58,'promotionCodes':[],'selectedPayment':'VISA','deliveryTotal':0,'shippingTax':0,'totalAdjustment':0}]
+dlo-debug.js:1136   [3] suffix output (numKeys=0 sizeOfValues=0 sizeOfPayload=34)
+  ['Order Completed',{'transactionID_str':'2325112','grandTotal_real':45.59,'subTotal_real':37.02,'shippingTotal_real':5.99,'totalTax_real':2.58,'promotionCodes_strs':[],'selectedPayment_str':'VISA','deliveryTotal_int':0,'shippingTax_int':0,'totalAdjustment_int':0}]
+
+Order Completed {transactionID_str: '2325112', grandTotal_real: 45.59, subTotal_real: 37.02, shippingTotal_real: 5.99, totalTax_real: 2.58, …}
+```
+
+## Source Selection
+
+As mentioned, `source` is an important aspect of a rule: it defines which data layer to observe or read an initial value from.   In version 1.0.x, every `source` must be a JavaScript object with name value pairs - for example `digitalData.user`.
+
+The `source` property uses a custom selector syntax.  It’s most often seen as dot notation such as `digitalData.cart`, which can be read as, "Find the cart object inside the digitalData object that exists on the page."  Selector syntax is however much more expressive to allow a variety of common use cases.
+
+| Use Case | Sample | Description |
```

Comment:
I really dig how Qualtrics presented similar information:
![image](https://user-images.githubusercontent.com/45576380/88972189-29ad6680-d283-11ea-8e1b-3194c47e17da.png)
Would it be a good idea to follow this pattern? This might mean "Use Case" is folded into the description, "Sample" becomes "Selector" and an "Example" column is added that shows the selector in action on a CEDDL